### ISSUE TITLE
Returns invalid input even if number is valid

### ISSUE THREAD

**Issue by jonjieviduya** [jackocnr/intl-tel-input#1314]
Im trying to validate the input. But unfortunately some numbers return the `Invalid number` error even if the number is active and valid here in our country. Please see my code below for reference.

**index.html**

```
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<title>Document</title>
	<link rel="stylesheet" href="assets/int-tel-input/css/intlTelInput.min.css">
	<style>
		.hide { display: none; }
	</style>
</head>
<body>

	<div class="container">
		<input type="text" id="phone">
		<span id="valid-msg" class="hide">✓ Valid</span>
		<span id="error-msg" class="hide"></span>
	</div>
	
	<script src="assets/jquery.min.js"></script>
	<script src="assets/int-tel-input/js/intlTelInput.min.js"></script>
	<script src="assets/controller.js"></script>
</body>
</html>
```

**controller.js**

```
$(document).ready(function() {

	var input = document.querySelector("#phone"),
	  errorMsg = document.querySelector("#error-msg"),
	  validMsg = document.querySelector("#valid-msg");

	// here, the index maps to the error code returned from getValidationError - see readme
	var errorMap = ["Invalid number", "Invalid country code", "Too short", "Too long", "Invalid number"];

	// initialise plugin
	var iti = window.intlTelInput(input, {
		initialCountry: "PH",
		onlyCountries: ["ph"],
		separateDialCode: true,
		utilsScript: "assets/int-tel-input/js/utils.js"
	});

	var reset = function() {
	  input.classList.remove("error");
	  errorMsg.innerHTML = "";
	  errorMsg.classList.add("hide");
	  validMsg.classList.add("hide");
	};

	// on blur: validate
	input.addEventListener('blur', function() {
	  reset();
	  if (input.value.trim()) {
	    if (iti.isValidNumber()) {
	      validMsg.classList.remove("hide");
	    } else {
	      input.classList.add("error");
	      var errorCode = iti.getValidationError();
	      errorMsg.innerHTML = errorMap[errorCode];
	      console.log(errorCode)
	      errorMsg.classList.remove("hide");
	    }
	  }
	});

	// on keyup / change flag: reset
	input.addEventListener('change', reset);
	input.addEventListener('keyup', reset);

})
```

Reference:
https://intl-tel-input.com/node_modules/intl-tel-input/examples/gen/is-valid-number.html

**Comment by jackocnr** [comment]
We use [libphonenumber](https://github.com/googlei18n/libphonenumber) for formatting, validation and generating placeholder numbers - if your issue relates to one of these things, please check [their test site](https://rawgit.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/demo-compiled.html) first and if you see the problem there please file an issue with them instead.

If it's working correctly in libphonenumber, the next step is to check that we're using [the latest version](https://github.com/google/libphonenumber/releases) of libphonenumber in the plugin - check the version history [here](https://github.com/jackocnr/intl-tel-input/commits/master/build/js/utils.js). If we need to upgrade, it only takes 5 minutes to do this yourself and submit a Pull Request - instructions here: [Updating to a new version of libphonenumber](https://github.com/jackocnr/intl-tel-input/blob/master/.github/CONTRIBUTING.md#updating-to-a-new-version-of-libphonenumber).

**Comment by jackocnr** [comment]
I've just released v17.0.17 with the latest version of libphonenumber - please see if that fixes your issue, else you'll need to open an issue with libphonenumber.
